### DISCUSSION TITLE
What is the minimum browser support for SolidJS?

### DISCUSSION THREAD

**Discussion by cyfung1031** [solidjs/solid#2165] [Q&A]
I find that it cannot be used in Waterfox Classic.

Would you please suggest the minimum browser requirement?

SolidJS playground

<img width="1508" alt="Screen Shot 2024-05-19 at 15 12 30" src="https://github.com/solidjs/solid/assets/44498510/0327d931-5c7f-43b0-b1fd-510a6cd9be3f">

**Comment by atk** [comment]
At least for stores, you need proxy object support: https://caniuse.com/proxy

**Comment by ryansolid** [comment]
If we were advertise minimum browser versions the newest features we leverage are `globalThis` and Proxies. A decent amount of stuff can be polyfilled, but not Proxies. But I think Chrome 48 is the lowest version that runs stock. I'm not sure about the others. The hard part is that Solid came out after the age of evergreen browsers so for 99% of users we could say the last 5 versions or something and be fine since proxies don't work in IE11 it was never supported. Basically every major browser created after 2015 supported Solid. Other lower power devices though have stricter constraints. We do have the ability to to run without proxies and my understanding is there are TV devices running on Chrome 38 with Solid, but I doubt something like the playground is built that strictly.

**Comment by chiefcll** [comment]
Just confirming that Solid is used on old TV devices. Here it is running on Chrome 38 which is before Proxy support. We accomplish this by avoiding splitProps and other areas which might require Proxies.

![Chrome38](https://github.com/user-attachments/assets/bb397459-38ac-42ba-afff-192ac69c4639)

Running on lambdatest.com

**Comment by JLarky** [comment]
I found a datapoint on this. If you hydrate your components then you are relying on `globalThis._$HY` beeing supported by the browser 
<img width="313" alt="image" src="https://github.com/user-attachments/assets/f5caa584-b202-4851-bfb6-7db494632af3" />

which puts it to Chrome 71+ accoding to [caniuse](https://caniuse.com/?search=globalThis)

**Comment by cyfung1031** [comment]
@ryansolid 

I got three errors when I use Waterfox Classic to navigate https://playground.solidjs.com/

Waterfox Classic has already supported `globalThis` and `Proxy` but still not working.

**Reply by ryansolid** [reply]
To be fair the playground itself uses a bunch of features newer than what Solid baseline supports(like web workers). If you want help clarification on the playground maybe I'd look at that repo.
